engine/src/flutter/shell/platform/darwin/ios/framework/Source/overlay_layer_pool.mm

```diff
@@ -23,12 +23,13 @@
                                    SkRect rect,
                                    int64_t view_id,
                                    int64_t overlay_id) {
-  // There can be a race where UpdateViewState() is called when flutter_view is nil when app is
-  // backgrounded.
-  if (!flutter_view) {
+  // There can be a race where UpdateViewState() is called when flutter_view or flutter_view's
+  // screen is nil when app is backgrounded.
+  FlutterView* flutterView = (FlutterView*)flutter_view;
+  if (!flutterView || !flutterView.screen || flutterView.screen.scale == 0.0f) {
```

Comment:
The scale is actually 0.0 when it's backgrounded?

Reply:
I don't think scale can be `0`, but just wanna be extra safe here.

Reply:
Yeah given this can result in a divide-by-zero, seems reasonable to check. Agreed it would seem like a definite UIKit bug to me if they ever returned a scale of 0, but checking for it seems fair.

You could consider a comment that we know `flutterView` and `flutterView.screen` may be nil, but given scale is used in a scaling calculation division below, we're being paranoid.